src/core/unit.c

```diff
@@ -5056,6 +5056,52 @@ static void unit_update_dependency_mask(Hashmap *deps, Unit *other, UnitDependen
                 assert_se(hashmap_update(deps, other, di.data) == 0);
 }
 
+void unit_remove_dependencies_of_type(Unit *u, UnitDependencyMask mask, UnitDependency d) {
+        bool done;
+
+        assert(d < _UNIT_DEPENDENCY_MAX);
+        assert(u);
+
+        /* Removes all dependencies of type d that u has on other units marked
+         * for ownership by 'mask'. */
+
+        if (mask == 0)
+                return;
+
+        do {
```

Comment:
Dependencies now are mapped via hashmap, given you have the type which is used as key, why not just using it instead of a full iteration, which sounds expensive?

Reply:
Makes sense, thanks. I'll re-work.

Reply:
Re-worked, we do still need to iterate on the back dependencies but we can get rid of the forward dependencies in 1 swoop.  PTAL!